sdk/core/core-client-rest/review/core-client.api.md

```diff
@@ -46,6 +46,20 @@ export type ClientOptions = PipelineOptions & {
 // @public
 export function createRestError(message: string, response: PathUncheckedResponse): RestError;
 
+// @public
+export interface ErrorModel {
+    code: string;
+    details: Array<ErrorModel>;
+    innererror?: InnerError;
```

Comment:
These models come from the cadl model definition https://github.com/Azure/cadl-azure/blob/main/packages/cadl-azure-core/lib/foundations.cadl#L87-L113. @xirzec do you think we should use cause in typescript to replace the innererror ?

Reply:
I'm not opposed to `cause`; I think @deyaaeldeen investigated this before

Reply:
The `cause` option is used in the [avro serializer](https://github.com/Azure/azure-sdk-for-js/blob/a9f4825c7c2d156b1f813360b0e6f1b057c288ee/sdk/schemaregistry/schema-registry-avro/src/avroSerializer.ts#L260-L269) and there're no issues with it so far with one catch, it is not supported by the TS types until recently, see https://github.com/microsoft/TypeScript/issues/45167 for context. So if you expect customers to be using the bleeding edge of TS, then that is not an issue. Otherwise, we might need to expose our own `Error`.

Reply:
Ok I understand the change better now, my only recommendation is to run the name discussion by the archboard because I believe `innerError` is already used in many SDKs, following their respective swaggers.

Reply:
I am not sure how do I feel about this. On one hand, `cause` is a recent addition to JS's `Error`, so in theory, it is a good and idiomatic name. On the other hand, I never seen it used by our services in their error models. Is this model going to be used to represent actual service errors? If yes, I am leaning towards reverting to the `innererror` name but I defer to @bterlson

Reply:
after talking with @deyaaeldeen  offline and compared with dotnet error type,  it's fine to keep the inner error.